tests/Builder/Expression/Pipelines.php

```diff
@@ -171,41 +490,705 @@ enum Pipelines: string
      */
-    case MergeObjectsMergeObjects = <<<'JSON'
+    case MapAddToEachElementOfAnArray = <<<'JSON'
     [
         {
-            "$lookup": {
-                "from": "items",
-                "localField": "item",
-                "foreignField": "item",
-                "as": "fromItems"
-            }
-        },
-        {
-            "$replaceRoot": {
-                "newRoot": {
-                    "$mergeObjects": [
-                        {
-                            "$arrayElemAt": [
-                                "$fromItems",
+            "$project": {
+                "adjustedGrades": {
+                    "$map": {
+                        "input": "$quizzes",
+                        "as": "grade",
+                        "in": {
+                            "$add": [
+                                "$$grade",
                                 {
-                                    "$numberInt": "0"
+                                    "$numberInt": "2"
                                 }
                             ]
-                        },
-                        "$$ROOT"
-                    ]
+                        }
+                    }
                 }
             }
-        },
+        }
+    ]
+    JSON;
+
+    /**
+     * Truncate Each Array Element
+     *
+     * @see https://www.mongodb.com/docs/manual/reference/operator/aggregation/map/#truncate-each-array-element
+     */
+    case MapTruncateEachArrayElement = <<<'JSON'
+    [
         {
             "$project": {
-                "fromItems": {
-                    "$numberInt": "0"
+                "city": "$city",
+                "integerValues": {
+                    "$map": {
+                        "input": "$distances",
+                        "as": "decimalValue",
+                        "in": {
+                            "$trunc": [
+                                "$$decimalValue"
+                            ]
+                        }
+                    }
+                }
+            }
+        }
+    ]
+    JSON;
+
+    /**
+     * Convert Celsius Temperatures to Fahrenheit
+     *
+     * @see https://www.mongodb.com/docs/manual/reference/operator/aggregation/map/#convert-celsius-temperatures-to-fahrenheit
+     */
+    case MapConvertCelsiusTemperaturesToFahrenheit = <<<'JSON'
+    [
+        {
+            "$addFields": {
+                "tempsF": {
+                    "$map": {
+                        "input": "$tempsC",
+                        "as": "tempInCelsius",
+                        "in": {
+                            "$add": [
+                                {
+                                    "$multiply": [
+                                        "$$tempInCelsius",
+                                        {
+                                            "$numberDouble": "1.8000000000000000444"
```

Comment:
Other example of numeric value incorrectly encoded.
Related to https://github.com/mongodb/mongo-php-builder/pull/21#discussion_r1448851990